yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -120,6 +121,8 @@ class YarnClusterSuite extends FunSuite with BeforeAndAfterAll with Matchers wit
     logInfo(s"RM address in configuration is ${config.get(YarnConfiguration.RM_ADDRESS)}")
 
     fakeSparkJar = File.createTempFile("sparkJar", null, tempDir)
+    hadoopConfDir = new File(tempDir, Client.HADOOP_CONF_DIR)
+    assert(hadoopConfDir.mkdir())
```

Comment:
The directory is a temp dir (so unique to this suite) and cleaned up when the JVM shuts down (through hooks set up by the Utils class). Is there anything else you were thinking of here?